src/test/java/com/internship/backend/service/ReservationServiceTest.java

```diff
@@ -1,12 +1,168 @@
 package com.internship.backend.service;
 
+import com.internship.backend.exceptions.InvalidDateException;
+import com.internship.backend.exceptions.ReservationAlreadyExists;
+import com.internship.backend.exceptions.ReservationDoesNotExistException;
+import com.internship.backend.exceptions.TennisCourtDoesNotExistsException;
+import com.internship.backend.model.NewDate;
+import com.internship.backend.model.Reservation;
+import com.internship.backend.model.TennisCourt;
+import com.internship.backend.repository.PriceRepository;
+import com.internship.backend.repository.ReservationRepository;
+import com.internship.backend.repository.TennisCourtRepository;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.InjectMocks;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
 import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
 
+@ExtendWith(MockitoExtension.class)
 class ReservationServiceTest {
 
+    @Mock
+    private ReservationRepository reservationRepository;
+
+    @Mock
+    private TennisCourtRepository tennisCourtRepository;
+
+    @Mock
+    private PriceRepository priceRepository;
+
+    @InjectMocks
+    private ReservationService reservationService;
+
+    @BeforeEach
+    void setUp() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    void addReservation_validReservation() throws InvalidDateException, ReservationAlreadyExists {
+        //arrange
+        Reservation reservation = new Reservation();
+        NewDate startTime = new NewDate(0, 17, 18, 7, 2024);
+        NewDate endTime = new NewDate(0, 19, 18, 7, 2024);
+        reservation.setStartTime(startTime);
+        reservation.setEndTime(endTime);
+
+        when(reservationRepository.findAll()).thenReturn(new ArrayList<>());
+        when(reservationRepository.save(reservation)).thenReturn(reservation);
+
+        //act
+        Reservation result = reservationService.addReservation(reservation);
+
+        //assert
+        assertNotNull(result);
+        assertEquals(reservation, result);
+    }
+
+    @Test
+    void addReservation_invalidReservation() throws InvalidDateException, ReservationAlreadyExists {
```

Comment:
This test could be renamed to something like: shouldNotSaveInvalidReservations ?